gtsam/hybrid/HybridGaussianConditional.cpp

```diff
@@ -254,12 +270,18 @@ std::shared_ptr<HybridGaussianFactor> HybridGaussianConditional::likelihood(
   const DiscreteKeys discreteParentKeys = discreteKeys();
   const KeyVector continuousParentKeys = continuousParents();
   const HybridGaussianFactor::FactorValuePairs likelihoods(
-      conditionals_,
-      [&](const GaussianConditional::shared_ptr &conditional)
-          -> GaussianFactorValuePair {
-        const auto likelihood_m = conditional->likelihood(given);
-        const double Cgm_Kgcm = conditional->negLogConstant() - negLogConstant_;
-        return {likelihood_m, Cgm_Kgcm};
+      factors(),
+      [&](const GaussianFactorValuePair &pair) -> GaussianFactorValuePair {
+        if (auto conditional =
+                std::dynamic_pointer_cast<GaussianConditional>(pair.first)) {
+          const auto likelihood_m = conditional->likelihood(given);
+          // scalar is already correct.
+          assert(pair.second ==
```

Comment:
Maybe we can remove this assert and save some performance?

Reply:
Done